extensions/incubator/src/main/java/io/opentelemetry/extension/incubator/trace/ExtendedBaggageBuilder.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.extension.incubator.trace;
+
+import io.opentelemetry.api.baggage.Baggage;
+import io.opentelemetry.api.baggage.BaggageBuilder;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import java.util.Map;
+
+public class ExtendedBaggageBuilder {
+  private final BaggageBuilder delegate;
+
+  private ExtendedBaggageBuilder(BaggageBuilder delegate) {
+    this.delegate = delegate;
+  }
+
+  public static ExtendedBaggageBuilder create(BaggageBuilder builder) {
+    return new ExtendedBaggageBuilder(builder);
+  }
+
+  public static ExtendedBaggageBuilder current() {
+    return create(Baggage.current().toBuilder());
+  }
+
+  public ExtendedBaggageBuilder set(String key, String value) {
+    delegate.put(key, value);
+    return this;
+  }
```

Comment:
This doesn't seem right. Why have a method that performs the same function with a different name?

Reply:
it's a different level of abstraction. "put" relates to `Map` - as a user, I don't case if it's stored in a Map.